integration/network_test.go

```diff
@@ -0,0 +1,861 @@
+				return err == nil
+			},
+		},
+		{
+			testName:          "Access HTTP server from different subnet with port access",
+			spName:            "network1",
+			serviceName:       "instance_different_subnet",
+			allowedConnection: []string{"http_server" + "/" + httpListenerPort},
+			expectedRunContainerStatus: func(err error) bool {
+				return err == nil
+			},
+		},
+		{
+			testName:    "Access HTTP server from different subnet without port access",
+			spName:      "network1",
+			serviceName: "instance_different_subnet_not_allowed_port",
+			expectedRunContainerStatus: func(err error) bool {
+				return err != nil
+			},
+		},
+	}
+
+	for _, tCases := range cases {
+		t.Run(tCases.testName, func(t *testing.T) {
+			containerPath := path.Join(tmpDir, tCases.serviceName)
+
+			if err := createOCIContainer(manager, containerPath, tCases.serviceName,
+				[]string{
+					"curl", strings.Join([]string{"http_server", httpListenerPort}, ":"),
+					"--connect-timeout", "2",
+				}, false); err != nil {
+				t.Errorf("Can't create container: %v", err)
+			}
+
+			if err := manager.AddInstanceToNetwork(tCases.serviceName, tCases.spName, networkmanager.NetworkParams{
+				Hosts:              []aostypes.Host{{IP: ipHTTPServer, Hostname: "http_server"}},
+				HostsFilePath:      path.Join(containerPath, "etc", "hosts"),
+				ResolvConfFilePath: path.Join(containerPath, "etc", "resolv.conf"),
+				AllowedConnections: tCases.allowedConnection,
+			}); err != nil {
+				t.Errorf("Can't add instance to network: %v", err)
+			}
+
+			if err := runOCIContainer(containerPath, tCases.serviceName); !tCases.expectedRunContainerStatus(err) {
+				t.Error("Unexpected result run container")
+			}
+		})
+	}
+
+	for _, tCases := range cases {
+		if err = manager.RemoveInstanceFromNetwork(tCases.serviceName, tCases.spName); err != nil {
+			t.Errorf("Can't remove instance from network: %v", err)
+		}
+	}
+}
+
+func TestNetworkDNS(t *testing.T) {
+	manager, err := networkmanager.New(&config.Config{WorkingDir: tmpDir}, nil)
+	if err != nil {
+		t.Fatalf("Can't create network manager: %v", err)
+	}
+	defer manager.Close()
+
+	casesInfinityRunningInstance := []struct {
+		spName       string
+		instanceName string
+		params       networkmanager.NetworkParams
+	}{
+		{
+			spName:       "network0",
+			instanceName: "serverDNS1",
+			params: networkmanager.NetworkParams{
+				Hostname: "server1",
+				Aliases:  []string{"DNS1"},
+				InstanceIdent: cloudprotocol.InstanceIdent{
+					Instance:  0,
+					SubjectID: "subject1",
+					ServiceID: "serverDNS1",
+				},
+			},
+		},
+		{
+			spName:       "network1",
+			instanceName: "server2",
+			params: networkmanager.NetworkParams{
+				Hostname: "serverDNS2",
+				Aliases:  []string{"DNS2"},
+				InstanceIdent: cloudprotocol.InstanceIdent{
+					Instance:  1,
+					SubjectID: "subject2",
+					ServiceID: "serverDNS2",
+				},
+			},
+		},
+	}
+
+	for _, tCase := range casesInfinityRunningInstance {
```

Comment:
here is better :) Can be used in all cases.

Reply:
Done